projects/components/src/splitter/splitter.component.ts

```diff
@@ -0,0 +1,294 @@
+/* eslint-disable max-lines */
+import {
+  AfterContentInit,
+  ChangeDetectionStrategy,
+  Component,
+  ContentChildren,
+  ElementRef,
+  EventEmitter,
+  Inject,
+  Input,
+  OnChanges,
+  Output,
+  QueryList,
+  Renderer2
+} from '@angular/core';
+import { LayoutChangeService, TypedSimpleChanges, queryListAndChanges$ } from '@hypertrace/common';
+import { EMPTY, Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { SplitterDirection } from './splitter';
+import { SplitterCellDimension, SplitterContentDirective } from './splitter-content.directive';
+import { DOCUMENT } from '@angular/common';
+import { debounce } from 'lodash-es';
+
+@Component({
+  selector: 'ht-splitter',
+  styleUrls: ['./splitter.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  template: `
+    <div class="splitter-container" [ngClass]="[this.direction | lowercase]">
+      <ng-container *ngFor="let content of this.contents$ | async as contents; let index = index">
+        <div class="splitter-content" [ngStyle]="{ flex: this.getFlex | htMemoize: content.dimension }">
+          <ng-container *ngTemplateOutlet="content.templateRef"></ng-container>
+        </div>
+
+        <div
+          class="splitter"
+          *ngIf="
```

Comment:
shouldn't the ngIf on `this.direction` should be on the parent div?

Reply:
Let me set a default value and move the ngIf